### PR TITLE
Remove Fedora 37 codepaths

### PR DESCRIPTION
**Pull request by supakeen**
This removes any code paths relating to checks that assume Fedora 37 exists, see #470.

### CODE REVIEW

File: pkg/distro/fedora/package_sets.go [hunk 1]
```diff
@@ -498,13 +497,7 @@ func iotInstallerPackageSet(t *imageType) rpmmd.PackageSet {
...
 	ps := anacondaPackageSet(t)
 
-	releasever := t.Arch().Distro().Releasever()
-	version, err := strconv.Atoi(releasever)
-	if err != nil {
-		panic("cannot convert releasever to int: " + err.Error())
-	}
-
-	if version >= 38 {
+	if !common.VersionLessThan(t.arch.distro.osVersion, "39") {
```
Comment (thozza) [inline pkg/distro/fedora/package_sets.go]: Just making sure that the intention here is to basically also change the condition to `version >= 39`, which is consistent with the previous commit and the rest of this file, but not mentioned explicitly in the commit message with this commit.

**Review by thozza — APPROVED** [review]
LGTM. I love the addition of `VersionGreaterThanOrEqual`, which makes reading the code much easier... :heart_eyes: I added one question, but I'm 99% sure that it is expected and intentional.

File: pkg/distro/fedora/package_sets.go (hunk 1 above)
Comment (supakeen) [reply pkg/distro/fedora/package_sets.go]: It is the intention to move this to make it consistent with the other version checks which were all `>= 39`.

**Comment by supakeen** [comment]
@thozza since #469 has landed I've rebased this PR and it should now be ready for merging :)

File: pkg/distro/rhel7/distro_test.go [hunk 2]
```diff
@@ -445,7 +445,7 @@ func TestDistroFactory(t *testing.T) {
 			expected: rhel7.DistroFactory("rhel-7.9"),
 		},
 		{
-			strID:    "fedora-37",
+			strID:    "fedora-38",
```
Comment (thozza) [inline pkg/distro/rhel7/distro_test.go:448]: There's a very little value in making these changes going forward, but it does not hurt for the sake of consistency... :slightly_smiling_face:

**Review by thozza — APPROVED** [review]
Thanks!

**Pull request merged**